testing/scenario_app/android/app/src/androidTest/java/dev/flutter/scenariosui/ScreenshotUtil.java

```diff
@@ -0,0 +1,257 @@
+  void init() {
+    if (mXmlSerializer != null) {
+      return;
+    }
+    album = AlbumImpl.create(Registry.getRegistry().instrumentation.getContext(), "default");
+    album.cleanup();
+
+    mXmlSerializer = Xml.newSerializer();
+    try {
+      mOutputStream =
+          new BufferedOutputStream(new FileOutputStream(album.getMetadataFile()), BUFFER_SIZE);
+    } catch (FileNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+    try {
+      mXmlSerializer.setOutput(mOutputStream, "utf-8");
+      mXmlSerializer.startDocument("utf-8", null);
+      // Start tag <screenshots>.
+      mXmlSerializer.startTag(null, "screenshots");
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  void writeText(String tagName, String value) throws IOException {
+    if (mXmlSerializer == null) {
+      throw new RuntimeException("ScreenshotUtil must be initialized.");
+    }
+    mXmlSerializer.startTag(null, tagName);
+    mXmlSerializer.text(value);
+    mXmlSerializer.endTag(null, tagName);
+  }
+
+  void writeBitmap(Bitmap bitmap, String name, String testClass, String testName)
+      throws IOException {
+    if (mXmlSerializer == null) {
+      throw new RuntimeException("ScreenshotUtil must be initialized.");
+    }
+    album.writeBitmap(name, 0, 0, bitmap);
+
+    mXmlSerializer.startTag(null, "screenshot");
+    writeText("name", name);
+    writeText("test_class", testClass);
+    writeText("test_name", testName);
+    writeText("tile_width", "1");
+    writeText("tile_height", "1");
+    mXmlSerializer.endTag(null, "screenshot");
+  }
+
+  /** Finishes metadata.xml. */
+  void flush() {
+    if (mXmlSerializer == null) {
+      throw new RuntimeException("ScreenshotUtil must be initialized.");
+    }
+    try {
+      // End tag </screenshots>
+      mXmlSerializer.endTag(null, "screenshots");
+      mXmlSerializer.endDocument();
+      mXmlSerializer.flush();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    try {
+      mOutputStream.close();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    album.flush();
+
+    mXmlSerializer = null;
+    mOutputStream = null;
+    album = null;
+  }
+
+  private static int getStatusBarHeight() {
+    final Context context = InstrumentationRegistry.getTargetContext();
+    final int resourceId =
+        context.getResources().getIdentifier("status_bar_height", "dimen", "android");
+    int statusBarHeight = 0;
+    if (resourceId > 0) {
+      statusBarHeight = context.getResources().getDimensionPixelSize(resourceId);
+    }
+    return statusBarHeight;
+  }
+
+  private static int getActionBarHeight(Activity activity) {
+    int actionBarHeight = 0;
+    final android.content.res.TypedArray styledAttributes =
+        activity.getTheme().obtainStyledAttributes(new int[] {android.R.attr.actionBarSize});
+    actionBarHeight = (int) styledAttributes.getDimension(0, 0);
```

Comment:
I wonder if we should get a rounded int instead of floor, as `getDimension` could get something like x.9999999?

Reply:
I don't think so. This value always appears to be an integer.